terraform/modules/cluster/addons.tf

```diff
@@ -1,35 +1,17 @@
-data "aws_eks_addon_version" "latest" {
-  for_each = toset(["vpc-cni"])
-
-  addon_name         = each.value
-  kubernetes_version = var.cluster_version
-  most_recent        = true
-}
-
-resource "aws_eks_addon" "vpc_cni" {
```

Comment:
Based on @niallthomson's findings, the EKS module was updated to allow configuring addons before compute is ready https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2478

In addition, there is now a variable `dataplane_wait_duration` within the module that users can set a variable time duration between the control plane becoming active, and the start of downstream compute resource provisioning. Currently it defaults to `"30s"`